src/tools/perf/lib/libperf.c

```diff
@@ -958,38 +980,31 @@ static void ucp_perf_release_requests_in_progress(ucx_perf_context_t *perf,
     }
 }
 
-static void ucp_perf_test_destroy_self_eps(ucx_perf_context_t *perf)
+static void
+ucp_perf_test_destroy_eps(ucx_perf_context_t *perf, ucp_ep_domain_t ep_domain)
 {
-    unsigned thread_count   = perf->params.thread_count;
-    unsigned num_in_prog    = 0;
-    ucs_status_ptr_t **reqs = ucs_alloca(thread_count * sizeof(*reqs));
-    ucs_status_ptr_t *req;
+    unsigned thread_count  = perf->params.thread_count;
+    unsigned num_in_prog   = 0;
+    ucs_status_ptr_t *reqs = ucs_alloca(thread_count * sizeof(*reqs));
+    ucs_status_ptr_t req;
     unsigned i;
 
-    for (i = 0; i < thread_count; ++i) {
-        ucp_rkey_h self_rkeys[] = {perf->ucp.tctx[i].perf.ucp.self_send_rkey,
-                                   perf->ucp.tctx[i].perf.ucp.self_recv_rkey};
-        req = ucp_perf_test_destroy_ep(perf->ucp.tctx[i].perf.ucp.self_ep,
-                                       self_rkeys, 2, i);
-        if (req != NULL) {
-            reqs[num_in_prog++] = req;
-        }
+#define UCP_RKEY_DESTROY(_rkey) \
+    if ((_rkey) != NULL) { \
+        ucp_rkey_destroy(_rkey); \
     }
 
-    ucp_perf_release_requests_in_progress(perf, reqs, num_in_prog);
-}
-
-static void ucp_perf_test_destroy_eps(ucx_perf_context_t *perf)
-{
-    unsigned thread_count   = perf->params.thread_count;
-    unsigned num_in_prog    = 0;
-    ucs_status_ptr_t **reqs = ucs_alloca(thread_count * sizeof(*reqs));
-    ucs_status_ptr_t *req;
-    unsigned i;
-
     for (i = 0; i < thread_count; ++i) {
-        req = ucp_perf_test_destroy_ep(perf->ucp.tctx[i].perf.ucp.ep,
-                                       &perf->ucp.tctx[i].perf.ucp.rkey, 1, i);
+        if (ep_domain & UCP_EP_DOMAIN_SELF) {
```

Comment:
why need this check, isn't it enough we're checking rkey!=NULL?

Reply:
so maybe split it to different functions, as it was before?

Reply:
Ok, agree, after all I see more code with new approach
Reverted